### ISSUE TITLE
Unused flags when adding TPM to VMs

### ISSUE THREAD

**Issue by hamistao** [canonical/lxd#13560]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 22.04
 * snap: latest/edge
 * LXC version: 5.21.1 LTS
 * LXD version: 5.21.1 LTS

# Issue description

When adding a TPM to a container it is possible to modify the device paths for the TPM and TPM resource manager via the flags `path` and `pathrm`. When doing the same with VMs, the flags are recognized by the CLI and they are added to the device config on LXD, but aside from that these flags' values aren't used for any practical purpose and the default paths `/dev/tpm0` and `/dev/tpmrm0` are used regargless of the flag values. This gives an illusion to the user that the device paths are configurable when in fact they aren't. If there isn't support in QEMU for customizing these paths I would say not allowing to use these flags with VMs would be an acceptable way to handle this since, considering that QEMU only allows one TPM device per VM, I can't see many use cases for this customization.

# Steps to reproduce
 1. `lxc init ubuntu:n vm --vm`
 2. `lxc config device add vm tpm-device tpm path=/dev/tpm-test0 pathrm=/dev/tpmrm-test0` 
 3. `lxc start vm`
 4. `lxc exec vm -- stat /dev/tpm0 # works`
 5. `lxc exec vm -- stat /dev/tpm-test0 # fails`

**Comment by tomponline** [comment]
I agree in principle, but have you considered the case where a tpm device is added to a profile that is used with both containers and VMs?

**Comment by hamistao** [comment]
@tomponline Good point, I hadn't considered that. And that is even trickier once one considers that containers require the paths to be explicitly defined.
